data_steward/cdr_cleaner/cleaning_rules/generalize_sex_gender_concepts.py

```diff
@@ -1,103 +1,206 @@
-
 MAN_CONCEPT_ID = 1585839
-
 SEX_AT_BIRTH_MALE_CONCEPT_ID = 1585846
-
 SEX_AT_BIRTH_FEMALE_CONCEPT_ID = 1585847
 
-
-def parse_query_for_updating_woman_to_generalized_concept_id(
-    project_id, dataset_id):
-    """
-    This function returns an update query to update the gender to the generalized gender concept_id for the cases
-    where the biological sex is reported as male and gender is reported as woman.
-    :param project_id: the project id
-    :param dataset_id: the dataset id
-    :return:an update query to update the gender from woman to the generalized concept id
-    """
-    return GENERALIZED_CONCEPT_ID_QUERY_TEMPLATE.format(
-        project_id=project_id,
-        dataset_id=dataset_id,
-        gender_value_source_concept_id=WOMAN_CONCEPT_ID,
-        biological_sex_birth_concept_id=SEX_AT_BIRTH_MALE_CONCEPT_ID,
-        generalized_gender_concept_id=GENERALIZE_GENDER_CONCEPT_ID)
-
-
-def parse_query_for_updating_man_to_generalized_concept_id(
-    project_id, dataset_id):
-    """
-    This function returns an update query to update the gender to the generalized gender concept_id for the cases
-    where the biological sex is reported as female and gender is reported as man.
-    :param project_id: the project id
-    :param dataset_id: the dataset id
-    :return:an update query to update the gender from man to the generalized concept id
-    """
-    return GENERALIZED_CONCEPT_ID_QUERY_TEMPLATE.format(
-        project_id=project_id,
-        dataset_id=dataset_id,
-        gender_value_source_concept_id=MAN_CONCEPT_ID,
-        biological_sex_birth_concept_id=SEX_AT_BIRTH_FEMALE_CONCEPT_ID,
-        generalized_gender_concept_id=GENERALIZE_GENDER_CONCEPT_ID)
-
-
-def get_generalized_concept_id_queries(project_id,
-                                       dataset_id,
-                                       sandbox_dataset_id=None):
-    """
-    This function generates a list of query dicts for updating the records for which we need to the generalize gender
-    in both of value_as_concept_id and value_source_concept_id
-    :param project_id: the project_id in which the query is run
-    :param dataset_id: the dataset_id in which the query is run
-    :param sandbox_dataset_id: Identifies the sandbox dataset to store rows 
-    #TODO use sandbox_dataset_id for CR
-    :return: a list of query dicts for updating the gender concept ids
-    """
-
-    queries = []
-
-    query = dict()
-    query[cdr_consts.
-          QUERY] = parse_query_for_updating_woman_to_generalized_concept_id(
-              project_id, dataset_id)
-    query[cdr_consts.BATCH] = True
-    queries.append(query)
-
-    query = dict()
-    query[cdr_consts.
-          QUERY] = parse_query_for_updating_man_to_generalized_concept_id(
-              project_id, dataset_id)
-    query[cdr_consts.BATCH] = True
-    queries.append(query)
-
-    return queries
+SANDBOX_CONCEPT_ID_QUERY_TEMPLATE = JINJA_ENV.from_string("""
+INSERT INTO
```

Comment:
This will need to be a `CREATE OR REPLACE TABLE` statement.  The `INSERT INTO` statement assumes a table already exists.  Most cleaning rules assume the table does not exist when the rule begins.

Reply:
Since the query will be called twice, I decided to create the sandbox table in the `setup_rule` method to not overwrite any sandbox records from the first run.